lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -112,25 +116,21 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   end
 
   defp project_doc(bin, module_id, locals, compiled) when is_binary(bin) do
-    %{
-      aliases: aliases,
-      docs_refs: docs_refs,
-      extension: extension,
-      lib_dirs: lib_dirs,
-      modules_refs: modules_refs
-    } = compiled
+    options =
+      compiled
+      |> Map.merge(%{
+        module_id: module_id,
+        locals: locals
+      })
+      |> Map.to_list()
```

Comment:
I don't like that we are converting structured list into unstructured one. My suggestion is to keep using a map internally everywhere.

Reply:
fixed